tests/mocha/astnode_test.js

```diff
@@ -316,51 +324,141 @@ suite('ASTNode', function() {
     });
 
     suite('In', function() {
-      test('block', function() {
-        var node = Blockly.ASTNode.createBlockNode(this.blocks.statementInput1);
-        var inNode = node.in();
-        var field = this.blocks.statementInput1.inputList[0].fieldRow[0];
-        assertEquals(inNode.getLocation(), field);
+      setup(function() {
+        Blockly.defineBlocksWithJsonArray([{
+          "type": "dummy_input",
+          "message0": "Hello",
+          "colour": 230,
+          "tooltip": "",
+          "helpUrl": ""
+        },
+        {
+          "type": "dummy_inputValue",
+          "message0": "Hello %1  %2",
+          "args0": [
+            {
+              "type": "input_dummy"
+            },
+            {
+              "type": "input_value",
+              "name": "NAME"
+            }
+          ],
+          "colour": 230,
+          "tooltip": "",
+          "helpUrl": ""
+        }]);
+        var dummyInput = this.workspace.newBlock('dummy_input');
+        var dummyInputValue = this.workspace.newBlock('dummy_inputValue');
+        var fieldWithOutput2 = this.workspace.newBlock('field_input');
+
+        this.blocks.dummyInput = dummyInput;
+        this.blocks.dummyInputValue = dummyInputValue;
+        this.blocks.fieldWithOutput2 = fieldWithOutput2;
+        this.emptyWorkspace = new Blockly.Workspace();
       });
-      test('input', function() {
+      teardown(function() {
+        delete this.blocks.dummyInput;
+        delete this.blocks.dummyInputValue;
+        delete this.blocks.fieldWithOutput2;
+        delete this.emptyWorkspace;
+      });
+
+      test('fromInputToOutput', function() {
         var input = this.blocks.statementInput1.inputList[0];
         var node = Blockly.ASTNode.createInputNode(input);
         var inNode = node.in();
         var outputConnection = this.blocks.fieldWithOutput.outputConnection;
         assertEquals(inNode.getLocation(), outputConnection);
       });
-      test('blockToInput', function() {
+      test('fromInputToNull', function() {
+        var input = this.blocks.statementInput2.inputList[0];
+        var node = Blockly.ASTNode.createInputNode(input);
+        var inNode = node.in();
+        assertEquals(inNode, null);
+      });
+      test('fromInputToNext', function() {
```

Comment:
Is this actually fromInputToPrevious?  It looks like it's going from a statement input to the previous connection on the first block in the substack.

Reply:
Yeah it should be fromInputToPrevious. Just changed the name